SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -77,35 +77,73 @@ class const_ref {
   static std::string get_description() { return "const reference"; }
 
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems> call_simd_ctor(const DataT *ref_data) {
+  static void call_simd_ctor(const DataT *ref_data, DataT *out) {
     simd<DataT, NumElems> source_simd;
     source_simd.copy_from(ref_data);
-    return call_simd_by_const_ref<DataT, NumElems>(
-        simd<DataT, NumElems>(source_simd));
+    call_simd_by_const_ref<DataT, NumElems>(simd<DataT, NumElems>(source_simd),
+                                            out);
   }
 
 private:
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems>
-  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref) {
-    return simd_by_const_ref;
+  static void
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
   }
 };
 
-int main(int argc, char **argv) {
-  sycl::queue queue{esimd_test::ESIMDSelector{},
-                    esimd_test::createExceptionHandler()};
+// Struct that calls simd in provided context and then verifies obtained result.
+struct Verifier {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  static bool verify(sycl::queue &queue, const std::vector<DataT> ref_data,
+                     const std::string &data_type) {
+    bool passed = true;
+
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>{queue});
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         shared_allocator<DataT>{queue});
+
+    queue.submit([&](sycl::handler &cgh) {
+      const auto ref = shared_ref_data.data();
+      auto out = result.data();
+
+      cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(ref, out);
+          });
+    });
+    queue.wait_and_throw();
+
+    for (size_t i = 0; i < result.size(); ++i) {
+      if (!are_bitwise_equal(ref_data[i], result[i])) {
+        passed = false;
+
+        auto test_description = TestDescription<DataT, NumElems>(
+            TestCaseT::get_description(), data_type, result[i], ref_data[i], i);
+        log::fail(test_description);
+      }
+    }
+
+    return passed;
+  }
+};
 
-  bool passed{true};
+int main(int argc, char **argv) {
```

Comment:
```suggestion
int main(int, char **) {
```
To avoid warning about unused variables. Here and elsewhere.

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/614/commits/e87ad2ec4b8cff0070719ae8992d50d244c2a019#diff-e20a89556dfa9ec98f8eec4bea698de3975b780ff24a3bde1ced890af1d16581L132-L133)

Reply:
Let's have it in a `ctors_default` as well

Reply:
oh, my mistake. was removed in this commit: e757cc8bfa62226cc21f28c8811647fed4c7cadb